pkg/analysis_server/lib/src/services/correction/fix_internal.dart

```diff
@@ -1875,6 +1884,29 @@ class FixProcessor {
     }
   }
 
+  /// This fix is not going to work well if what you want is to remove only a
+  /// TypeName inside a FunctionTypedFormalParameter (returnType, parameter or
+  /// generic) so if you want to do that you should create other fix or modify
+  /// this one carefully.
```

Comment:
Does this comment (and the lack of a test method for it) mean that the test I suggested fails? If so, I consider that to be a blocker. One possible way to get around that is to pass a `bool` into this method that controls whether it looks for `FunctionTypedFormalParameter`s.